rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -32,23 +32,18 @@ RSocketStateMachine::RSocketStateMachine(
     std::shared_ptr<RSocketStats> stats,
     std::shared_ptr<RSocketConnectionEvents> connectionEvents)
     : mode_(mode),
-      resumeCache_(std::make_shared<ResumeCache>(stats)),
-      streamState_(std::make_shared<StreamState>(*stats)),
+      stats_(stats ? stats : RSocketStats::noop()),
+      resumeCache_(stats_),
```

Comment:
With cold-resumption, resumeCache will be constructed and passed in by the client-application.  Keeping it a smart-pointer will make it convenient on the application part (if the application also wants access to it)